src/vm/memory/segments.zig

```diff
@@ -135,6 +149,15 @@ pub const MemorySegmentManager = struct {
         return self.memory.data.count();
     }
 
+    /// Retrieves the number of temporary memory segments.
+    ///
+    /// # Returns
+    ///
+    /// The number of temporary memory segments as a `usize`.
+    pub fn numTempSegments(self: *Self) usize {
```

Comment:
Can we add a unit test for this function in order to be complete?

Reply:
Done!

it's not a good pattern to add to the hash map directly using `put` because nothing stops me from adding a segment with a negative index to the `data` and not `temp_data`.